react/states_and_effects/more_on_state.md

```diff
@@ -51,11 +51,15 @@ function Person() {
 }
 ~~~
 
+
 <div class="lesson-note lesson-note--tip" markdown="1">
 
-#### Nested objects and arrays in state
+#### Objects and arrays in state
+
+In the above example, notice how we _create_ a new object, and then copy the existing state values into the new object while providing a new value for `age`.
+That is because, besides the unpredictability that comes up from changing the state manually, if we don't provide a new object to `setState` it is not guaranteed to re-render the page, this is intentional for optimization. Therefore we should always provide a new Object for `setState` to trigger a re-render. `setState` uses [Object.is](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is) to determine if the previous state is the same.
```

Comment:
I'm not sure what you mean here. This sentence answers the `why does setState skip re-rendering even if we pass the same object?` The `why` that helps optimization is just implementation details at this point.

I can also remove it since my last commit Link talks about Optimization as well, but that was an optional read, I wasn't really sure where to fit them all together.

Reply:
I meant only the "this is intentional for optimization" The whole sentence seems great other than this part which I think can be left out

Reply:
I might just do that if you think it enhanced the readability of the whole paragraph.